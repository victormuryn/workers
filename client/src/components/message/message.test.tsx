import React from 'react';
import renderer from 'react-test-renderer';
import Message from './message';

describe(`Message tests`, () => {
  it(`Message danger renders corrects`, () => {
    const tree = renderer.create(<Message
      text="Test text for test"
      onClose={() => {}}
      type="danger"
    />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Message without type renders corrects`, () => {
    const tree = renderer.create(<Message
      text="Test text for test"
      onClose={() => {}}
    />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
