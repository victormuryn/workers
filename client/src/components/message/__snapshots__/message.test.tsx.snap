// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message tests Message danger renders corrects 1`] = `
<div
  className="msg-container"
>
  <div
    className="xd-message msg-danger"
  >
    <div
      className="xd-message-icon"
    >
      <svg
        viewBox="0 0 30 10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          className="ion-alert"
          href="#icon-danger"
        />
      </svg>
    </div>
    <div
      className="xd-message-content"
    >
      <p>
        Test text for test
      </p>
    </div>
    <a
      className="xd-message-close"
      href="#"
      onClick={[Function]}
    >
      <svg
        viewBox="0 0 30 10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          className="close-icon"
          href="#icon-danger"
        />
      </svg>
    </a>
  </div>
</div>
`;

exports[`Message tests Message without type renders corrects 1`] = `
<div
  className="msg-container"
>
  <div
    className="xd-message msg-info"
  >
    <div
      className="xd-message-icon"
    >
      <svg
        viewBox="0 0 30 10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          className="ion-alert"
          href="#icon-info"
        />
      </svg>
    </div>
    <div
      className="xd-message-content"
    >
      <p>
        Test text for test
      </p>
    </div>
    <a
      className="xd-message-close"
      href="#"
      onClick={[Function]}
    >
      <svg
        viewBox="0 0 30 10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          className="close-icon"
          href="#icon-danger"
        />
      </svg>
    </a>
  </div>
</div>
`;
